flume-ng-channels/flume-kafka-channel/src/main/java/org/apache/flume/channel/kafka/KafkaChannel.java

```diff
@@ -296,8 +321,9 @@ private synchronized ConsumerAndRecords createConsumerAndRecords() {
       KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<String, byte[]>(consumerProps);
       ConsumerAndRecords car = new ConsumerAndRecords(consumer, channelUUID);
       logger.info("Created new consumer to connect to Kafka");
-      car.consumer.subscribe(Arrays.asList(topic.get()),
```

Comment:
I'm not sure why you've changed this.

Reply:
Why are we adding in a dangling ChannelRebalanceListener. Can we be sure that this doesn't introduce bugs?

Reply:
@tmgstevens, references to ChannelRebalanceListener has been removed.

Reply:
Still confused here. It looks like you've removed the `ChannelRebalanceListener` altogether. I don't think we need to do this. Please add this back in.